### DISCUSSION TITLE
macos install-homebrew.sh how to bypass inputing password

### DISCUSSION THREAD

**Discussion by zoezhangmattr** [actions/virtual-environments#10175] [Image configuration]
try to build macos runner image by using runner-images scripts, 
https://github.com/actions/runner-images/blob/0ad636b8243987e85a0148adcfd379b1661a0728/images/macos/templates/macOS-14.arm64.anka.pkr.hcl#L172
it asks for inputing sudo password
the user can do sudo, 
how does github bypass this prompt? thanks

**Comment by pravinade** [comment]
Hi @zoezhangmattr,
 
We have tested the bypass by inputting the password, and it worked as expected on our end. Additionally, I noticed that the sudo password prompt did not appear when running sudo brew list during the workflow run. This suggests that the sudoers configuration is allowing commands to run without requiring a password. Please find the code snippet below for your reference.

<img width="1277" alt="Screenshot 2024-11-27 at 11 27 35 AM" src="https://github.com/user-attachments/assets/c884c8b1-47ac-4329-9eaf-352b38bd1278">

<img width="1784" alt="Screenshot 2024-11-27 at 11 26 38 AM" src="https://github.com/user-attachments/assets/cd993382-7d26-40e4-a5cd-4af34efe8ffe">
